api/src/test/java/bio/terra/tanagra/service/query/BinaryColumnFilterTest.java

```diff
@@ -0,0 +1,321 @@
+  // each test method
+  private final EntityMapping.Builder entityMappingProtoBuilder =
+      EntityMapping.newBuilder()
+          .setEntity("entityA")
+          .setPrimaryKey(
+              ColumnId.newBuilder()
+                  .setDataset("datasetA")
+                  .setTable("tableA")
+                  .setColumn("columnA")
+                  .build());
+
+  @Test
+  @DisplayName(
+      "underlay conversion fails when binary column filter table doesn't match the entity pk table")
+  void filterColumnAndPrimaryKeyFromDifferentTables() {
+    TableFilter tableFilterProto =
+        TableFilter.newBuilder()
+            .setBinaryColumnFilter(
+                BinaryColumnFilter.newBuilder()
+                    .setColumn(
+                        ColumnId.newBuilder()
+                            .setDataset("datasetA")
+                            .setTable("tableB")
+                            .setColumn("columnC")
+                            .build())
+                    .setOperator(BinaryColumnFilterOperator.LESS_THAN)
+                    .setInt64Val(45)
+                    .build())
+            .build();
+
+    Exception exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> convertUnderlayFromProto(tableFilterProto),
+            "exception thrown when binary column filter table doesn't match entity primary key table");
+    LOG.info("underlay conversion exception message: {}", exception.getMessage());
+    assertTrue(
+        exception
+            .getMessage()
+            .contains(
+                "Binary table filter column is not in the same table as the entity primary key"),
+        "exception message says that binary column filter table doesn't match entity primary key table");
+  }
+
+  @Test
+  @DisplayName(
+      "underlay conversion fails when binary column filter value doesn't match the column data type")
+  void filterColumnAndValueHaveDifferentDataTypes() {
+    TableFilter tableFilterProto =
+        TableFilter.newBuilder()
+            .setBinaryColumnFilter(
+                BinaryColumnFilter.newBuilder()
+                    .setColumn(
+                        ColumnId.newBuilder()
+                            .setDataset("datasetA")
+                            .setTable("tableA")
+                            .setColumn("columnA")
+                            .build())
+                    .setOperator(BinaryColumnFilterOperator.LESS_THAN)
+                    .setInt64Val(64)
+                    .build())
+            .build();
+
+    Exception exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> convertUnderlayFromProto(tableFilterProto),
+            "exception thrown when binary column filter value doesn't match the column data type");
+    LOG.info("underlay conversion exception message: {}", exception.getMessage());
+    assertTrue(
```

Comment:
nit: suggest removing the log and using the more semantic substring assertion (throughout), similar to this: https://github.com/DataBiosphere/tanagra/blob/10c982b25f5fd437dd884b90fe57c9600db6031b/api/src/test/java/bio/terra/tanagra/service/query/api/ApiConversionServiceTest.java#L75

Personally I have a slight preference for the Truth library, but this is the current pattern.

Reply:
done.